code/model/modelreplace.cpp

```diff
@@ -0,0 +1,409 @@
+extern void create_family_tree(polymodel* obj);
+extern void model_calc_bound_box(vec3d* box, vec3d* big_mn, vec3d* big_mx);
+
+// General Functions and external code paths
+
+bool model_exists(const SCP_string& filename) {
+	auto it = virtual_pofs.find(filename);
+	if (it != virtual_pofs.end() && !it->second.empty()) {
+		return true;
+	}
+
+	return cf_exists_full(filename.c_str(), CF_TYPE_MODELS);
+}
+
+bool read_virtual_model_file(polymodel* pm, const SCP_string& filename, model_parse_depth depth, int ferror, model_read_deferred_tasks& deferredTasks) {
+	auto virtual_pof_it = virtual_pofs.find(filename);
+
+	//We don't have a virtual pof
+	if(virtual_pof_it == virtual_pofs.end())
+		return false;
+
+	//We have one, but we're already past it and are processing whatever it overwrote
+	int& depthLocal = depth[filename];
+
+	if ((int)virtual_pof_it->second.size() <= depthLocal)
+		return false;
+
+	const auto& virtual_pof = virtual_pof_it->second[depthLocal];
+	depthLocal++;
+
+	read_model_file(pm, filename.c_str(), ferror, deferredTasks, depth);
+
+	for (const auto& operation : virtual_pof.operationList)
+		operation->process(pm, deferredTasks, depth);
+
+	return true;
+}
+
+// Parsing function
+
+static void parse_virtual_pof() {
+	SCP_string name;
+	stuff_string(name, F_NAME);
+
+	auto& virtual_pof_list = virtual_pofs[name];
+	virtual_pof_list.emplace_back();
+	VirtualPOFDefinition& toFill = virtual_pof_list.back();
+
+	toFill.name = name;
+	
+	required_string("+Base POF:");
+	stuff_string(toFill.basePOF, F_FILESPEC);
+
+	do {
+		ignore_white_space();
+		char operation_type[NAME_LENGTH];
+		stuff_string(operation_type, F_NAME, NAME_LENGTH);
+
+		auto operation_parser = virtual_pof_operations.find(operation_type);
+		if (operation_parser != virtual_pof_operations.end())
+			toFill.operationList.emplace_back(operation_parser->second());
+		else {
+			error_display(1, "Unknown operation type %s in virtual POF %s!", operation_type, name.c_str());
+			virtual_pof_list.pop_back();
+			skip_to_start_of_string_either("$POF:", "#End");
+			return;
+		}
+	} while (optional_string_either("$POF:", "#End") == -1);
+	Mp -= 5;
+}
+
+static void parse_virtual_pof_table(const char* filename) {
+	try
+	{
+		read_file_text(filename, CF_TYPE_TABLES);
+		reset_parse();
+
+		required_string("#Virtual POFs");
+
+		while (optional_string("$POF:"))
+			parse_virtual_pof();
+
+		required_string("#End");
+	}
+	catch (const parse::ParseException& e)
+	{
+		mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", filename, e.what()));
+		return;
+	}
+}
+
+void virtual_pof_init() {
+	if (cf_exists_full("virtual_pofs.tbl", CF_TYPE_TABLES))
+		parse_virtual_pof_table("virtual_pofs.tbl");
+
+	parse_modular_table(NOX("*-pof.tbm"), parse_virtual_pof_table);
+}
+
+// Internal helper functions
+
+#define REPLACE_IF_EQ(data, source, dest) if ((data) == source) (data) = dest;
```

Comment:
Considering that this is a macro, and that all the occurrences are used with parameters called `source` and `dest`, you could rewrite the macro to only require the first parameter -- or alternatively add a second macro that calls the first